### DISCUSSION TITLE
Adaptive Firefox Color Synced with Website Colors

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#864] [Ideas]
### What feature would you like to see?

Currently I am using an extension called "Adaptive Tab Bar Color" but when I installed it, it didn't work. It would be great to have better integration with that or a full featured in browser implementation of it.

### What platform are you seeing the problem on?

Linux

### Additional context

Adaptive Tab Bar Color: 
https://addons.mozilla.org/en-US/firefox/addon/adaptive-tab-bar-colour/

I have been using this configuration for my firefox browser to get a simple look with sidebery. It'd be great to have a sidebery integration as well.
https://gitlab.com/scientiac/tesseract.home.manager/-/tree/main/apps/graphical/firefox

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/295_

**Comment by SomeDudeBrosky** [comment]
This would be an amazing feature

**Comment by shrimpwtf** [comment]
this would be awesome

**Comment by stripedgoat** [comment]
What happens when a workspace has already a color set by the user? It would clash with it?

**Comment by fvnky07** [comment]
any updates?

**Comment by fvnky07** [comment]
I just found something out. I installed the extension and found that the color of my browser slightly shifted. but then i opened another application  and focused it with my mouse and my zen browser's inactive colors became the adaptive tab colors.

Really odd but it does work just not in the right way...
It only works when you have focused another window.

**Comment by owendor** [comment]
Any updates to this? I would love this to be natively supported in Zen - it's a native feature on Chromium-based browsers and it's a really nice touch.

**Comment by BlaiteHere** [comment]
this pls <3 and ill switch to zen

**Comment by DIMFLIX** [comment]
Any updates on this topic? The only thing I'm missing to fully switch to Zen Browser is a dynamic sidebar color based on the page content color.

**Reply by fvnky07** [reply]
yeah it would be pretty good but i havent found anything in terms of update. im sure it could be vibe coded in a pr maybe. but i just switch the sidebar to black as its the most neutral color for browsing the web probably.

**Reply by Red-Pumpkin20** [reply]
@DIMFLIX idk if this is still relevant but I found two mods that achieve this result (both are installable through Sine)
https://github.com/kkugot/blended-addressbar - this one blends just the toolbar if you use the sidebar + toolbar mode
https://github.com/caezium/zen-page-tint - this one does the full tinting for both the sidebar and the toolbar

**Reply by DIMFLIX** [reply]
Thank you, yes, that's relevant. I'll check when I get home.

**Reply by englut** [reply]
Not sure what Sine is, but do y'all have anything that can be installed via Firefox extensions or the Zen mods store?

**Reply by Red-Pumpkin20** [reply]
@englut Sine is a third party mod store for Firefox-based browsers. It's safe to use if you are worried about that and quite easy to install. In terms of Firefox extensions there was this one https://addons.mozilla.org/en-US/firefox/addon/adaptive-tab-bar-colour/ but I'm not sure it works on Zen

**Reply by englut** [reply]
@Red-Pumpkin20 can you link the Sine mod? I found one (which I think might be the one you're referring to), but there are issues regarding virus detection triggering. Which makes me think the mod store install I found is either the wrong one, or it takes a hacky approach that borderlines as malware.

**Reply by Red-Pumpkin20** [reply]
@englut here's the link to the official Sine mod store github repo https://github.com/CosmoCreeper/Sine
As for the Sine mods that I've mentioned they are in the quoted reply

> @DIMFLIX idk if this is still relevant but I found two mods that achieve this result (both are installable through Sine) https://github.com/kkugot/blended-addressbar - this one blends just the toolbar if you use the sidebar + toolbar mode https://github.com/caezium/zen-page-tint - this one does the full tinting for both the sidebar and the toolbar
